### PR TITLE
Improve key parsing

### PR DESCRIPTION
**Pull request by NathanReb**
A few improvements both not to accept invalid keys and to give better feedback about those to users.

### CODE REVIEW

File: key-tester.js [hunk 1]
```diff
@@ -126,6 +126,7 @@ function run(form_url, ga_code) {
             $.getJSON(url, function(data) {
                 var key = data[0].key;
                 $("#form-ssh-key").val(key);
+                check_ssh_key();
```
Comment (emillon) [inline key-tester.js]: `check_ssh_key` is already called in the `.change` handler so you can also trigger it (`$('#form-ssh-key').trigger('change')`), not sure if it's better though.

File: key-tester.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ function run(form_url, ga_code) {
         });
         $('#ssh-form').submit(function(e) {
             e.preventDefault();
-            var key = $('#form-ssh-key').val();
+            var key = rsa.sanitize($('#form-ssh-key').val());
```
Comment (emillon) [inline key-tester.js]: Are there occasions where you would not want to sanitize a key? If that's not the case, how about just making it a private function that you call during `rsa.parse`?

File: rsa.js [hunk 3]
```diff
@@ -20,26 +20,72 @@ function sanitize(key) {
...
+function removeEndingEquals(blob) {
+  if (blob.endsWith('==')) {
+    return blob.slice(0,-2);
+  } else if (blob.endsWith('=')) {
+    return blob.slice(0,-1);
+  }
+  return blob;
+}
+
+function illegalBase64(blob) {
```
Comment (emillon) [inline rsa.js]: I think that checking that the blob matches `[A-Za-z0-9/+]+={0,2}` would considerably simplify this piece of code while being good enough (you lose the distinction between "bad padding" and "wrong character" but if the key is invalid there's nothing much you can do anyway).

File: key-tester.js (hunk 1 above)
Comment (NathanReb) [inline key-tester.js]: I think I prefer to trigger a change, I'll make the change !

File: rsa.js (hunk 3 above)
Comment (NathanReb) [inline rsa.js]: I agree that it makes the code too complex only to slightly improve error reporting to the user. I'll remove it !

File: key-tester.js (hunk 2 above)
Comment (NathanReb) [inline key-tester.js]: In fact sanitize is already used within `rsa.parse`. This is a mistake as I wanted to sanitize the key before sending it to formkeep but that's not the way. I'll fix this.

**Comment by NathanReb** [comment]
Pushed some changes ! Let me know what you think

**Comment by emillon** [comment]
Looks good to me, thanks!

**Pull request merged**